File: client/travel/src/app/core/dto/TourResponseDto.ts
import { TourStatus } from '../models/Tour';

export interface TourResponseDto {
  id: number;
  title: string;
  description: string;
  price: number;
  startDate: string;
  endDate: string;
  status: TourStatus;
  thumbnailImageId: number;
  destinationId: number;
  destinationName: string;
  destinationCountry: string;
  reviewsCount: number;
  reviewsRatingAverage: number;
}

File: api/travel/src/main/java/com/example/travel/dto/TourResponseDto.java
package com.example.travel.dto;

import com.example.travel.model.TourStatus;

import java.math.BigDecimal;

public record TourResponseDto(
        Long id,
        String title,
        String description,
        BigDecimal price,
        String startDate,
        String endDate,
        TourStatus status,
        Long thumbnailImageId,
        Long destinationId,
        String destinationName,
        String destinationCountry,
        Integer reviewsCount,
        Float reviewsRatingAverage
) {
}
